18F/identity-idp
# LG-8833: Support simulated reCAPTCHA results in test environments

## 🎫 Ticket

[LG-8833](https://cm-jira.usa.gov/browse/LG-8833)

## 🛠 Summary of changes

Adds a new mock implementation of the reCAPTCHA verifier and permits the assignment of a custom score to be used for evaluation.

## 📜 Testing Plan

Setup:

1. In `config/application.yml`:
   2. Set `phone_setup_recaptcha_score_threshold` to a number between `0.0` and `1.0`

Testing:

1. Go to http://localhost:3000
2. Create an account
3. Choose Phone as MFA when prompted
4. Observe "reCAPTCHA score" field
5. Enter a phone number (e.g. 5135550100 for U.S., 3065550100 for international)
6. Enter a simulated score value, considering the score threshold you assigned during setup
7. Submit
8. Observe that you are prompted for reCAPTCHA spam protection if (1) you entered an international number and (2) the simulated score you assigned in step 6 was lower than the threshold you assigned during setup
9. If presented spam protection screen, click "Continue" and observe that you are continued to the OTP confirmation

## 👀 Screenshots

![localhost_3000_add_phone](https://user-images.githubusercontent.com/1779930/220661960-bcf078b4-f1d2-4c50-a039-8019964728b2.png)
